bofire/strategies/predictives/botorch.py

```diff
@@ -332,91 +360,187 @@ def _optimize_acqf_continuous(
+            ic_gen_kwargs=ic_gen_kwargs,
+            nonlinear_constraints=nonlinear_constraints,  # type: ignore
+            fixed_features=fixed_features,
+            fixed_features_list=fixed_features_list,
+        )
+
+        # check if we should take the local step
+        if self.local_search_config.is_local_step(  # type: ignore
+            local_acqf_val.item(), global_acqf_val.item()
+        ):
+            return self._postprocess_candidates(candidates=local_candidates)
+
+        # otherwise we return a set of experiments that take us to the global
+        # candidates from the last experiment ran on the equipment
+        sp = ShortestPathStrategy(
+            data_model=ShortestPathStrategyDataModel(
                 domain=self.domain,
-                n_candidates=candidate_count,
-            )
-            candidates, acqf_vals = optimize_acqf(
-                acq_function=acqfs[0],
-                bounds=bounds,
-                q=candidate_count,
-                num_restarts=self.num_restarts,
-                raw_samples=self.num_raw_samples,
-                equality_constraints=get_linear_constraints(
-                    domain=self.domain,
-                    constraint=LinearEqualityConstraint,
-                )
-                + interpoints,
-                inequality_constraints=get_linear_constraints(
-                    domain=self.domain,
-                    constraint=LinearInequalityConstraint,
-                ),
-                fixed_features=fixed_features,
-                nonlinear_inequality_constraints=nonlinear_constraints,  # type: ignore
-                return_best_only=True,
-                options=self._get_optimizer_options(),  # type: ignore
-                ic_generator=ic_generator,
-                **ic_gen_kwargs,
+                start=self.experiments.iloc[-1].to_dict(),
+                end=self._postprocess_candidates(candidates).iloc[-1].to_dict(),
             )
-        return candidates, acqf_vals
+        )
+        step = pd.DataFrame(sp.step(sp.start)).T
+        return pd.concat((step, self.predict(step)), axis=1)
 
     def _ask(self, candidate_count: int) -> pd.DataFrame:  # type: ignore
-        """[summary]
+        """Ask the strategy for new experiments.
 
         Args:
-            candidate_count (int, optional): [description]. Defaults to 1.
-
-        Returns:
-            pd.DataFrame: [description]
-
+            candidate_count (int): The batch size of new experiments
+                requested from the strategy. Defaults to 1.
         """
         assert candidate_count > 0, "candidate_count has to be larger than zero."
         if self.experiments is None:
             raise ValueError("No experiments have been provided yet.")
 
+        if (
+            self.trust_region_config is not None
+            and not self.trust_region_config.has_sufficient_experiments(
+                self.experiments, self.domain
+            )
+        ):
```

Comment:
Can you add a comment to describe in natural language what is the scenario here, and what we are doing in this case?

Reply:
Essentially if you're new best solution sits at the edge of the old optimization box you might not have a sufficient number of points to fit a reasonable surrogate. In the original TurBO work I believe that the model was still fit on all the data in the history of a given trust region so this wasn't an issue. I am partial to the local model idea in MORBO where you take experiments within 2L of the reference point to train on (this will scale better in high-throughoput experimental settings as M^3 <<< N^3 for M<N). In that instance you want to sobol or otherwise sample a new set of experiments until you have enough to trust a local surrogate. 

This is a part where bofire wasn't that friendly because bofire validates that you return the requested number of candidates whereas I would want to return all the missing experiments in one shot.

Reply:
Ah ok, should we then also add a flag to the `TurBOConfig` that toggles between using local models for the TRs or one global model? This would be nice as we could then reproduce the literature to see if it actually works and gives even more flexibility. 

You are right regarding the bofire friendliness here, we could add a configuration flag to toggle this validation behavior.

Reply:
We can toggle the behaviour depending on `fit_region_multiplier` being `None`